docs/updating-the-docs.md

```diff
@@ -30,3 +30,21 @@ README](https://github.com/opensafely/documentation#building-locally-and-testing
 that details the use of `pip-compile` for this.
 
 ---8<-- 'includes/glossary.md'
+
+## Making changes to the data definition examples
```

Comment:
note: I started rewriting this section slightly in the `steve/databuilder-examples` branch. There's just an extra commit there.